File: ViewModels/ContentViewModel.cs
// Licensed to the .NET Foundation under one or more agreements.

using Octokit;

namespace FileSwap.ViewModels;

public class ContentViewModel
{
    public string? Name { get; set; }
    public StringEnum<ContentType> ContentType { get; set; }
    public String? Sha { get; set; }
    public string? Path { get; set; }
    public List<ContentViewModel>? Child { get; set; }

}

File: ClientApp/src/app/api/models/content-view-model.ts
/* tslint:disable */
/* eslint-disable */
import { ContentTypeStringEnum } from './content-type-string-enum';
export interface ContentViewModel {
  child?: null | Array<ContentViewModel>;
  contentType?: ContentTypeStringEnum;
  name?: null | string;
  path?: null | string;
  sha?: null | string;
}
